assets/src/edit-story/components/panels/colorPreset/index.js

```diff
@@ -0,0 +1,277 @@
+    },
+    actions: { updateStory, updateElementsById },
+  } = useStory();
+
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  const ref = useRef();
+
+  const {
+    actions: { showColorPickerAt, hideSidebar },
+  } = useSidebar();
+
+  const openColorPicker = useCallback(
+    (evt) => {
+      evt.stopPropagation();
+      showColorPickerAt(ref.current, {
+        onChange: () => null,
+        hasGradient: true,
+        hasOpacity: true,
+        onClose: hideSidebar,
+        addActions: getColorPickerActions,
+      });
+    },
+    [showColorPickerAt, hideSidebar]
+  );
+
+  const handleDeleteColor = useCallback(
+    (toDelete) => {
+      const colors = colorPresets.filter((color) => color !== toDelete);
+      updateStory({
+        properties: {
+          colorPresets: colors,
+        },
+      });
+    },
+    [colorPresets, updateStory]
+  );
+
+  const handleApplyBackgroundColor = useCallback(
+    (color) => {
+      updateElementsById({
+        elementIds: selectedElementIds,
+        properties: (currentProperties) => {
+          const { type } = currentProperties;
+          // @todo Is this necessary?
+          const { isMedia } = getDefinitionForType(type);
+          return isMedia
+            ? {}
+            : {
+                backgroundColor: color,
+              };
+        },
+      });
+    },
+    [selectedElementIds, updateElementsById]
+  );
+
+  const isText =
+    selectedElements.length > 0 &&
+    selectedElements.every(({ type }) => 'text' === type);
+
+  const handleApplyTextColor = useCallback(
+    (color) => () => {
+      if (isText) {
+        updateElementsById({
+          elementIds: selectedElementIds,
+          properties: { color },
+        });
+      }
+    },
+    [isText, selectedElementIds, updateElementsById]
+  );
+
+  const getSecondaryActions = () => {
+    return !isEditMode ? (
+      <>
+        <EditModeButton
+          onClick={(evt) => {
+            evt.stopPropagation();
+            setIsEditMode(true);
+          }}
+        >
+          <Edit />
+        </EditModeButton>
+        <AddColorPresetButton ref={ref} onClick={openColorPicker}>
+          <Add />
+        </AddColorPresetButton>
+      </>
+    ) : (
+      <ExitEditMode
+        onClick={(evt) => {
+          evt.stopPropagation();
+          setIsEditMode(false);
+        }}
+      >
+        {__('Exit', 'web-stories')}
+      </ExitEditMode>
+    );
+  };
+
+  return (
+    <Panel name="colorpreset">
+      <PanelTitle
+        isPrimary
+        secondaryAction={getSecondaryActions()}
+        displayCollapse={!isEditMode}
+      >
+        {__('Color presets', 'web-stories')}
+      </PanelTitle>
+      {colorPresets && (
+        <PanelContent isPrimary>
+          <Colors>
+            {colorPresets.map((color) => {
+              return (
+                <>
+                  <ButtonWrapper>
+                    <BackgroundColor
+                      {...generatePatternStyles(color)}
```

Comment:
This works for gradients too?

Reply:
It did not :) Will change that, however, since gradients can't be used for text color then not sure how to display that for the Text block. In theory, the Text block should have 2 versions of each color preset, one for text and one for the background. 🤷‍♀ Will need to check with UX.